From 66371e0386d5ab709988725841a42c86a9b4dcf1 Mon Sep 17 00:00:00 2001
From: velsinki <112010-velsinki@users.noreply.gitlab.gnome.org>
Date: Sun, 10 Dec 2023 23:31:17 +0100
Subject: [PATCH 4/4] plugins/gdk-pixbuf: remove lcms2 and revert to v0.8.2 API

lcms2 is not needed anymore due to the last patch. However, the last
patch was made with the newer, unreleased API. Using it with v0.8.2
therefore requires adding back a call parameter.
---
 plugins/gdk-pixbuf/CMakeLists.txt     | 2 +-
 plugins/gdk-pixbuf/pixbufloader-jxl.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/plugins/gdk-pixbuf/CMakeLists.txt b/plugins/gdk-pixbuf/CMakeLists.txt
index b15c48b7..b6b8a3b2 100644
--- a/plugins/gdk-pixbuf/CMakeLists.txt
+++ b/plugins/gdk-pixbuf/CMakeLists.txt
@@ -23,7 +23,7 @@ set_target_properties(pixbufloader-jxl PROPERTIES
 
 # Note: This only needs the decoder library, but we don't install the decoder
 # shared library.
-target_link_libraries(pixbufloader-jxl jxl jxl_threads lcms2 PkgConfig::Gdk-Pixbuf)
+target_link_libraries(pixbufloader-jxl jxl jxl_threads PkgConfig::Gdk-Pixbuf)
 
 execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} gdk-pixbuf-2.0 --variable gdk_pixbuf_moduledir --define-variable=prefix=${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE GDK_PIXBUF_MODULEDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
 install(TARGETS pixbufloader-jxl DESTINATION "${GDK_PIXBUF_MODULEDIR}")
diff --git a/plugins/gdk-pixbuf/pixbufloader-jxl.c b/plugins/gdk-pixbuf/pixbufloader-jxl.c
index 24c26e7b..fd74e5fb 100644
--- a/plugins/gdk-pixbuf/pixbufloader-jxl.c
+++ b/plugins/gdk-pixbuf/pixbufloader-jxl.c
@@ -416,6 +416,7 @@ static gboolean load_increment(gpointer context, const guchar *buf, guint size,
         JxlColorEncoding color_encoding;
         if (JXL_DEC_SUCCESS == JxlDecoderGetColorAsEncodedProfile(
                                    decoder_state->decoder,
+                                   &decoder_state->pixel_format,
                                    JXL_COLOR_PROFILE_TARGET_ORIGINAL,
                                    &color_encoding)) {
           // we don't check the return status here because it's not a problem if
-- 
2.43.0

